packages/paste-core/components/combobox/src/multiselect/MultiselectCombobox.tsx

```diff
@@ -0,0 +1,309 @@
+      (changes) => {
+        if (onSelectedItemsChangeProp) {
+          // Only pass the selectedItems, not the other downshift state data
+          onSelectedItemsChangeProp(changes.selectedItems);
+        }
+      },
+      [onSelectedItemsChangeProp]
+    );
+
+    const {
+      // Call optionally with preventKeyAction and destructure its returned object on the toggle button (select) or input (combobox) element.
+      getDropdownProps,
+      // Action prop that removes the item from the selection. Used in the X icon/button onClick handler.
+      removeSelectedItem,
+      addSelectedItem,
+      selectedItems,
+    } = useMultiSelectPrimitive<Item>({
+      initialSelectedItems,
+      onSelectedItemsChange,
+    });
+
+    const {
+      // State value with the open state of the menu. Used below for conditionally showing the items.
+      isOpen,
+      // Call and destructure its returned object on the toggle button (if any).
+      getToggleButtonProps,
+      // Call and destructure its returned object on the label element.
+      getLabelProps,
+      // Call and destructure its returned object on the menu element.
+      getMenuProps,
+      // Call and destructure its returned object on the input element.
+      getInputProps,
+      // Call and destructure its returned object on the input and toggle button parent.
+      getComboboxProps,
+      // Call with index or item and destructure its returned object on each menu item element.
+      getItemProps,
+      // State value with the index of thehighlighted menu item. Used below for styling.
+      highlightedIndex,
+    } = useComboboxPrimitive({
+      // The controlled state of currently available items to select
+      items,
+      /* Items are objects and not strings.
+       * As a result, the itemToString prop is passed to useCombobox. It will return
+       * the string equivalent of the item which will be used for displaying the item
+       * in the input once selected and for the a11y aria-live message that will occur
+       * on every item selection: ${itemToString(item)} has been selected.
+       *
+       * item.title is chosen to be the string equialent of each item object,
+       * so our prop will be passed as itemToString: item => item ? item.title : ''.
+       *
+       * Since clearing the input by Escape key is also considered an element change,
+       * we will return an empty string in this case.
+       */
+      itemToString,
+      // For multiselect, we need `selectedItem` to always be null because
+      // we use the stateChange event to add/remove items to the multiselect hook
+      selectedItem: null,
+      onInputValueChange,
+      onHighlightedIndexChange,
+      onIsOpenChange,
+      initialIsOpen,
+      // https://www.downshift-js.com/use-combobox#state-reducer
+      // Handles how state in Downshift should change as a result of user action
+      stateReducer(_state, actionAndChanges) {
```

Comment:
question: When we talked about override downshifts control over aria-selected on the dropdown listbox options, is this where we'd override it to `false` unless it's a selected item that appears in the pill list?

Reply:
I found a simpler approach that wouldn't be prone to Downshift's API changing by manually controlling it after the destructuring of downshift props on the ComboboxListboxOption component.